Improbable/Stdlib/Improbable.Stdlib/Platform/Platform.cs

```
@@ -0,0 +1,236 @@
+
+namespace Improbable.Stdlib.Platform
+{
+    public static class Platform
+    {
+        public static string ToolbeltConfigDir =>
+            Environment.GetEnvironmentVariable("IMPROBABLE_CONFIG_DIR") ?? (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                ? Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), ".improbable")
+                : Path.Combine(Environment.GetEnvironmentVariable("HOME"), ".improbable"));
+
+        private static PlatformCredential GetCredentials()
+        {
+            var refreshToken = File.ReadAllText(Path.Combine(ToolbeltConfigDir, "oauth2/oauth2_refresh_token"));
+            return new PlatformRefreshTokenCredential(refreshToken);
+        }
+
+        private static PlatformApiChannel GetLocalApiChannel()
+        {
+            return new PlatformApiChannel(GetCredentials(), new PlatformApiEndpoint("localhost", 9876, true));
+        }
+
+        public static Deployment StartLocal(in StartDeploymentOptions startDeployment)
+        {
+            var client = DeploymentServiceClient.Create(GetLocalApiChannel());
+
+            Shell.Run("spatial", "auth", "login", "--json_output");
+
+            // The runtime creates a '.improbable' folder next to the spatialos.json file.
+            var rootDir = Path.GetDirectoryName(startDeployment.ProjectConfigPath);
+            var tempDir = Path.Combine(rootDir, ".improbable");
+
+            Directory.CreateDirectory(tempDir);
+
+            var splConfigPath = Path.Combine(tempDir, "spl_config.json");
+            Shell.Run(
+                "spatial", "alpha", "config", "convert", "--json_output", "--launch_config", Shell.Escape(startDeployment.LaunchConfigPath),
+                "--main_config", Shell.Escape(startDeployment.ProjectConfigPath), "--output_path", splConfigPath);
+
+            var project = JsonConvert.DeserializeObject<ProjectConfig>(File.ReadAllText(startDeployment.ProjectConfigPath, Encoding.UTF8));
```

Probably worth try-catching to handle breaking changes in the FPL config files?

---

I'd just let exceptions bubble out